PyInstaller/building/build_main.py

```diff
@@ -219,8 +220,8 @@ def __init__(self, scripts, pathex=None, binaries=None, datas=None,
             # Create a Python module which contains the decryption key which will
             # be used at runtime by pyi_crypto.PyiBlockCipher.
             pyi_crypto_key_path = os.path.join(CONF['workpath'], 'pyimod00_crypto_key.py')
-            with open(pyi_crypto_key_path, 'w') as f:
-                f.write('key = %r\n' % cipher.key)
+            with open_file(pyi_crypto_key_path, 'w', encoding='utf-8') as f:
+                f.write(text_type('key = %r\n' % cipher.key))
```

Comment:
Because, in Python 2, `type('key = %r\n' % cipher.key)` is `<type 'str'>`, not `<type 'unicode'>`. That is, in Python 2, the *default* type is `binary_type`, not `text_type`. **Aaargh, Python 2 is so painful for Unicode!**

We could use `u'key = %r\n' % cipher.key`, but this makes removing this later (when we drop Python 2 support) harder.